pkg/kubelet/containerdshim/containerd_container.go

```diff
@@ -65,77 +96,68 @@ func (cs *containerdService) CreateContainer(podSandboxID string, containerConfi
 	if name == "" {
 		return "", fmt.Errorf("CreateContainerRequest.ContainerConfig.Name is empty")
 	}
-	attempt := containerConfig.GetMetadata().Attempt
-	var containerName string
-	if name == "infra" {
-		containerName = fmt.Sprintf("%s-%s", podSandboxID, name)
-	} else {
-		containerName = fmt.Sprintf("%s-%s-%v", podSandboxID, name, attempt)
-	}
-	containerID := podSandboxID // mikebrow TODO containerID must be unique crio guys are using stringid.GenerateNonCryptoID() then insuring uniqueness with storage
+	// mikebrow TODO containerID must be unique crio guys are using stringid.GenerateNonCryptoID() then insuring uniqueness with storage
+	containerId := dockershim.MakeContainerName(sandboxConfig, containerConfig)
 
-	tmpDir, err := getTempDir(containerName)
+	containerDir, err := ensureContainerDir(containerId)
 	if err != nil {
 		return "", err
 	}
 
-	rootfsPath := "/tmp/rootfs" // mikebrow TODO must be passed in
-	abs, err := filepath.Abs(rootfsPath)
-	if err != nil {
+	// Create container rootfs
+	rootfsPath := filepath.Join(containerDir, "rootfs")
+	repo, tag := repoAndTag(containerConfig.GetImage().GetImage())
```

Comment:
I believe different rootfses of the same image are sharing layers, there shouldn't be too much performance problem, right? Or did I misunderstand something?